src/middleware.ts

```diff
@@ -236,7 +269,9 @@ export const persist = <S extends State>(
     )
   }
 
-  const setItem = async () => {
+  const thenableSerialize = toThenable(serialize)
+
+  const setItem = (): Thenable<void> | void => {
```

Comment:
Does it return`void`? ~I'm fine with inferring return type (= no explicit return type), because we don't use it anywhere.~

Reply:
~Per my another comment, this may become just `void` (or no explicit typing).~

Reply:
Then this should simply be:
```suggestion
  const setItem = (): Thenable<void> => {
```

Reply:
I think we should remove `Thenable<void>` and `return` in `line:286` as well. what do you think?

Reply:
oh, `line:338` use it.